packages/devtools_app/lib/src/ui/flutter/service_extension_widgets.dart

```diff
@@ -202,120 +202,230 @@ class HotRestartButton extends StatelessWidget {
 
   @override
   void initState() {
     super.initState();
     // Only show the button if the device supports the given service.
-    _subscription = serviceManager.hasRegisteredService(
-      widget.serviceDescription.service,
-      (registered) {
+    autoDispose(
+      serviceManager.hasRegisteredService(
+        widget.serviceDescription.service,
+        (registered) {
+          setState(() {
+            _hidden = !registered;
+          });
+        },
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_hidden) return const SizedBox();
+
+    return InkWell(
+      onTap: () => invokeAndCatchErrors(widget.action),
+      child: Container(
+        constraints: const BoxConstraints.tightFor(width: 48.0, height: 48.0),
+        alignment: Alignment.center,
+        // TODO(djshuckerow): Just make these icons the right size to fit this box.
+        // The current size is a little tiny by comparison to our other
+        // material icons.
+        child: getIconWidget(widget.serviceDescription.icon),
+      ),
+    );
+  }
+}
+
+/// Checkbox that toggles the value of [structuredErrors].
+class StructuredErrorsCheckbox extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return _ServiceExtensionCheckbox(
+      service: structuredErrors,
+      describeError: (error) =>
+          'Failed to update structuredError settings: $error',
+    );
+  }
+}
+
+class _ServiceExtensionCheckbox extends _ServiceExtensionWidget {
+  const _ServiceExtensionCheckbox(
+      {Key key, this.service, @required String Function(dynamic) describeError})
+      : super(
+            key: key,
+            // Don't show messages on success or when this toggle is in progress.
+            completedText: null,
+            inProgressText: null,
+            describeError: describeError);
+  final ToggleableServiceExtensionDescription service;
+  @override
+  _ServiceExtensionMixin<_ServiceExtensionWidget> createState() =>
+      _ServiceExtensionCheckboxState();
+}
+
+class _ServiceExtensionCheckboxState extends State<_ServiceExtensionCheckbox>
+    with _ServiceExtensionMixin, AutoDisposeBase, AutoDisposeMixin {
+  bool checked = false;
+  @override
+  void initState() {
+    super.initState();
+    autoDispose(serviceManager.serviceExtensionManager.getServiceExtensionState(
+      widget.service.extension,
+      (data) {
         setState(() {
-          _hidden = !registered;
+          checked = data.value == widget.service.enabledValue;
         });
       },
-    );
+    ));
   }
 
   @override
-  void dispose() {
-    super.dispose();
-    _subscription.cancel();
+  Widget build(BuildContext context) {
+    return Tooltip(
+      message: checked
+          ? widget.service.enabledTooltip
+          : widget.service.disabledTooltip,
+      waitDuration: const Duration(seconds: 1),
+      child: InkWell(
+        onTap: _onClick,
+        child: Row(
+          children: <Widget>[
```

Comment:
remove unnecessary type

Reply:
The autocompletion for the Row constructor is doing this one automatically.  Type resolution in the analyzer has improved significantly since it was added.  We should see if we can turn it off.

@scheglov Konstantin, do you know if there is a special completion for Flutter Row and Column widgets in the Analyzer, or if this is coming from Visual Studio Code?

Reply:
This happens in IntelliJ, too, so I would assume it comes from a common source.

Reply:
It is coming from the analyzer. I agree it would be nice to omit the <Widget> to match current dart style.